### ISSUE TITLE
OverlayPanel and Sidebar: Not dismissable

### ISSUE THREAD

**Issue by Audiopolis** [primefaces/primereact#3422]
### Describe the bug

Prior to 8.0.0, OverlayPanel is dismissable (whether by default or by enabling `dismissable`). In every minor version of `>=8.0.0`, the OverlayPanel component does not hide when the outside is clicked. The prop to show the X button also has no effect (on the DOM or otherwise).

The behavior is the same in both Firefox and Chrome.

The OverlayPanel _does_ hide if the window/client area is resized in any way, but not if clicking on any other component in view.

Because I don't see this reported yet and it's been a little while since 8.0.0, I assume some other thing in my environment is affecting it.

At the same time as upgrading PrimeReact, I also upgraded:

```
    "@changey/react-leaflet-markercluster": "^4.0.0-rc1",
    "@reduxjs/toolkit": "^1.8.5",
    "@xstate/react": "^3.0.1",
    "leaflet": "^1.9.2",
    "leaflet.markercluster": "^1.5.3",
    "primereact": "^8.6.1",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-leaflet": "<4.1.0",
    "react-redux": "^8.0.4",
    "react-scripts": "5.0.1",
    "react-webcam": "^7.0.1",
    "redux": "^4.2.0",
```

The only broken thing is the OverlayPanel as far as I can tell.

Downgrading other dependencies (or PrimeReact) is not an option because these dependency versions are so co-dependent, unfortunately.

On further inspection, it seems like a bunch of other OverlayPanels work, but this one does not. There's no memoization or anything to explain it. They're all toggled using `overlayPanelRef.current?.toggle(e)` on a click. I decided to temporarily use a `Sidebar` instead, but it does the same thing. This particular Sidebar can't be hidden once opened, even if clicking directly on the X button. I compared the HTML resulting from two different OverlayPanels and two different Sidebars, and the HTML is identical in the same place in the DOM. The right one works, the left one does not:

![image](https://user-images.githubusercontent.com/22317842/193988061-4aea4df3-528d-4598-addd-19f6c34d1a9a.png)

The only difference is which button is clicked to show the OverlayPanel/Sidebar.

I gave the overlay (that should close the OverlayPanel/Sidebar when clicked) a red background and verified that the whole screen (minus the OverlayPanel/Sidebar) becomes red. I also tried increasing the z-index to match the working one.

One difference I found between OverlayPanel and Sidebar: Sidebar won't hide when the window is resized.

It's definitely pretty bizarre, and I understand that this might be hard to reproduce.

### PrimeReact version

8.6.1

### React version

18.x

### Language

ES6

### Build / Runtime

Create React App (CRA)

### Browser(s)

Chrome 106

### Steps to reproduce the behavior

I was unable to reproduce this as a minimal example, but this is the behavior:

In <8.0.0 (works):

1. Show an OverlayPanel when a div is clicked
2. In the OverlayPanel, add some `Checkbox` components
3. Make the OverlayPanel appear, then click elsewhere

Result: The OverlayPanel disappears.

In >=8.0.0 (does not work):

1. Show an OverlayPanel when a div is clicked
2. In the OverlayPanel, add some `Checkbox` components
3. Make the OverlayPanel appear, then click elsewhere

Result: The OverlayPanel stays where it is.

Additional steps:

- Resize the window: The OverlayPanel disappears
- Show/hide the console: The OverlayPanel disappears
- Interact with other components in some meaningful way: The OverlayPanel still stays

### Expected behavior

OverlayPanel/Sidebar closes when clicking outside it.

**Comment by melloware** [comment]
Are you saying you can't reproduce this in a code sandbox?

**Comment by Audiopolis** [comment]
Yes

**Comment by melloware** [comment]
Hmmm without a reproducible test case not sure how I can debug this one?

**Comment by Audiopolis** [comment]
I know :(

**Comment by Audiopolis** [comment]
So, I finally changed to a `Dialog` component. At first, it behaved the same, but adding an `onHide` handler and making it controlled (`visible={showDialog} onHide={() => setShowDialog(false)}`) fixes it. Tried with the `OverlayPanel` as well, and making it controlled fixes it, vs. using `overlayPanelRef.toggle`. Not sure if that is helpful, but if not, feel free to close, because my problem is fixed.

It's strange that making it controlled fixes it, and that downgrading to <8.0.0 fixes it. It seems the problem only affects uncontrolled components.

I hope that's at least somewhat useful. Unfortunately, it's the only information I have.

Thank you for your hard work!
